chef-config/lib/chef-config/workstation_config_loader.rb

```diff
@@ -138,6 +146,34 @@ def working_directory
       a
     end
 
+    def apply_credentials(creds, _profile)
+      Config.node_name = creds.fetch("node_name") if creds.key?("node_name")
+      Config.node_name = creds.fetch("client_name") if creds.key?("client_name")
```

Comment:
I wonder if this is worthy of a warning log message or similar if the user has both a `node_name` and a `client_name` in their credentials.

Reply:
Yeah that really needs a raise.